scrapy/core/downloader/__init__.py

```diff
@@ -99,28 +98,33 @@ def _deactivate(response):
     def needs_backout(self):
         return len(self.active) >= self.total_concurrency
 
-    def _get_slot(self, request, spider):
-        key = self._get_slot_key(request, spider)
-        if key not in self.slots:
-            conc = self.ip_concurrency if self.ip_concurrency else self.domain_concurrency
-            conc, delay = _get_concurrency_delay(conc, spider, self.settings)
-            self.slots[key] = Slot(conc, delay, self.randomize_delay)
-
-        return key, self.slots[key]
+    def _enqueue_request(self, request, spider):
+        d = defer.maybeDeferred(self._get_slot_key, request)
+        d.addCallback(self._slot_request, request, spider)
+        return d
 
-    def _get_slot_key(self, request, spider):
+    def _get_slot_key(self, request):
```

Comment:
I wonder if this may break some code - what's the reason for this change?

Reply:
The reason was simply cleanup: this is a private method and the implementation (both old and new) does not use the spider argument.

I don’t mind leaving it in.

Although, in that case, we might want to consider to make the method public.

Reply:
Do we make the method public, or do we leave it as is?